cli.js

```diff
@@ -1,84 +1,132 @@
+    process.exit(1);
+  }
+  await specificHandler(options);
+  process.exit(0);
 };
 
-commander
-  .version(version)
-  .description('Binaris command line interface')
-  .option('-p, --path <path>',
-    // eslint-disable-next-line quotes
-    'Use directory dir. "create" will create this directory if needed.');
-
-commander
-  .command('login')
-  .description('Login to your Binaris account using an API key')
-  .action(actionWrapper(loginHandler));
-
-commander
-  .command('logs <function>')
-  .description('Print the logs of a function')
-  .option('-t, --tail',
-    'Outputs logs in "tail -f" fashion')
-  .action(namedActionHandler(logHandler));
-
-commander
-  .command('create')
-  .description('Create a function from template')
-  .option('-f, --function <name>',
-    'Name of the function to generate. If omitted, a name will be chosen at random')
-  .action(actionWrapper(createHandler));
+yargs
+  .option('path', {
+    alias: 'p',
+    describe: 'Use directory dir. "create" will create this directory if needed.',
+    type: 'string',
+    global: true,
+  })
+  .usage(
+`Binaris command line interface
 
-commander
-  .command('deploy <function>')
-  .description('Deploys a function to the cloud')
-  .action(namedActionHandler(deployHandler));
+Usage: $0 <command> [options]` // eslint-disable-line comma-dangle
+  )
+  .command('create [function] [options]', 'Create a function from template', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 create [function] [options]')
+      .positional('function', {
+        describe: 'Name of the function to generate',
+        type: 'string',
+      })
+      .option('random', {
+        alias: 'r', describe: 'Generate a random name for your function', type: 'boolean',
+      });
+  }, async (argv) => {
+    if (argv.random) {
+      // eslint-disable-next-line no-param-reassign
+      argv.function = makeNameValid(moniker.choose());
+    }
+    await handleCommand(argv, createHandler);
+  })
+  .command('deploy <function> [options]', 'Deploys a function to the cloud', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 deploy <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to deploy',
+        type: 'string',
+      });
+  }, async (argv) => {
+    await handleCommand(argv, deployHandler);
+  })
+  .command('remove <function> [options]', 'Remove a previously deployed function', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 remove <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to remove',
+        type: 'string',
+      });
+  }, async (argv) => {
+    await handleCommand(argv, removeHandler);
+  })
+  .command('invoke <function> [options]', 'Invoke a Binaris function', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 invoke <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to invoke',
+        type: 'string',
+      })
+      .option('json', {
```

Comment:
We're going to need stding support for piping json pretty soon. i.e. `cat file.json | bn invoke foo`

Reply:
Shouldn't be an issue(but should be an _issue_)